torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -785,6 +787,15 @@ c10::intrusive_ptr<RRef> RRefContext::delForkOfOwner(
   return deletedRRef;
 }
 
+void RRefContext::blockCurrentStreams(
+    const std::vector<c10::Event>& events) const {
+  if (currentStreamFactory_) {
+    for (const c10::Event& event : events) {
+      event.block(currentStreamFactory_(event.device_index()));
+    }
+  }
```

Comment:
I think this could be also written as follows:
```suggestion
  for (const c10::Event& event : events) {
    c10::Device device{event.device_type(), event.device_index()};
    c10::Stream stream = c10::getDeviceGuardImpl(device.type())->getStream(device);
    event.block(stream);
  }
```
This way it leverages the c10 virtual device guards, which means it will work in a generic way both when CUDA is present and when it isn't. This allows us to avoid all this `currentStreamFactory_` complexity (moreover, before if `currentStreamFactory_` was null we'd silently avoid synchronizing the streams leading to correctness issues, whereas now if anything went wrong we'd see it immediately).

Here's some references on the above functions:
https://github.com/pytorch/pytorch/blob/7b31ba470897c4e5c4acb01d632ac71dd0b7d6ef/c10/core/impl/DeviceGuardImplInterface.h#L265
https://github.com/pytorch/pytorch/blob/7b31ba470897c4e5c4acb01d632ac71dd0b7d6ef/c10/core/impl/DeviceGuardImplInterface.h#L103